src/components/text-input/TextInput.js

```diff
@@ -29,14 +29,15 @@ class TextInput extends Component {
 
     renderInput = () => {
         const { placeholder, type, lineType, lineStrength } = this.props;
-        const currentLevel = typeof lineStrength !== 'undefined' && this.computeLevel();
+        const currentLevel = (typeof lineStrength !== 'undefined' && lineStrength >= 0 && lineStrength <= 1) && this.computeLevel();
```

Comment:
It makes sense for this to be calculated by the StrengthIndicator and not by this component. Can you quickly move it there?